edx_notifications/channels/tests/test_channel.py

```diff
@@ -0,0 +1,205 @@
+        'options': {
+            'display_name': 'channel_default',
+            'display_description': 'channel_description_default',
+        }
+    },
+    'channel1': {
+        'class': 'edx_notifications.channels.durable.BaseDurableNotificationChannel',
+        'options': {
+            'display_name': 'channel_name1',
+            'display_description': 'channel_description1',
+        }
+    },
+    'channel2': {
+        'class': 'edx_notifications.channels.durable.BaseDurableNotificationChannel',
+        'options': {
+            'display_name': 'channel_name2',
+            'display_description': 'channel_description2',
+        }
+    },
+    'channel3': {
+        'class': 'edx_notifications.channels.durable.BaseDurableNotificationChannel',
+        'options': {
+            'display_name': 'channel_name3',
+            'display_description': 'channel_description3',
+        }
+    },
+    'channel4': {
+        'class': 'edx_notifications.channels.durable.BaseDurableNotificationChannel',
+        'options': {
+            'display_name': 'channel_name4',
+            'display_description': 'channel_description4',
+        }
+    },
+    'channel5': {
+        'class': 'edx_notifications.channels.durable.BaseDurableNotificationChannel',
+        'options': {
+            'display_name': 'channel_name5',
+            'display_description': 'channel_description5',
+        }
+    },
+    'channel6': {
+        'class': 'edx_notifications.channels.durable.BaseDurableNotificationChannel',
+        'options': {
+            'display_name': 'channel_name6',
+            'display_description': 'channel_description6',
+        }
+    },
+}
+
+# list all of the mappings of notification types to channel
+_NOTIFICATION_CHANNEL_PROVIDER_TYPE_MAPS = {
+    '*': 'default',  # default global mapping
+    'edx_notifications.*': 'channel1',
+    'edx_notifications.channels.*': 'channel2',
+    'edx_notifications.channels.tests.*': 'channel3',
+    'edx_notifications.channels.tests.test_channel.*': 'channel4',
+    'edx_notifications.channels.tests.test_channel.channeltests.*': 'channel5',
+    'edx_notifications.channels.tests.test_channel.channeltests.foo': 'channel6'
+}
+
+
+@override_settings(NOTIFICATION_CHANNEL_PROVIDERS=_NOTIFICATION_CHANNEL_PROVIDERS)
+@override_settings(NOTIFICATION_CHANNEL_PROVIDER_TYPE_MAPS=_NOTIFICATION_CHANNEL_PROVIDER_TYPE_MAPS)
+class ChannelTests(TestCase):
+    """
+    Tests for channel.py
+    """
+
+    def setUp(self):
+        """
+        Harnessing
+        """
+        reset_notification_channels()
+        self.test_user_id = 1001  # an arbitrary user_id
+        self.test_msg_type = NotificationType(
+            name='edx_notifications.channels.tests.test_channel.channeltests.foo',
+            renderer='foo.renderer',
+        )
+
+    def cleanUp(self):  # pylint: disable=invalid-name
+        """
+        Test tear down
+        """
+        reset_notification_channels()
```

Comment:
Rather than a `cleanUp` method, you should be calling `self.addCleanup(reset_notification_channels)` in `setUp`. The equivalent method is `tearDown`, but using `addCleanup` is better (it has better guarantees around actually calling the cleanup method.